docs/dev/developer-guides/contracts/contracts.md

```diff
@@ -14,11 +14,17 @@ Summary:
 - **How deploying contracts works on zkSync.**
   To deploy a contract, a user calls the `create` function of the [ContractDeployer](./system-contracts.md#contractdeployer) and provides the hash of the contract to be published, as well as the constructor arguments. The contract bytecode itself is supplied in the `factory_deps` field of the EIP712 transactions. If the contract is a factory (i.e. it can deploy other contracts), these contracts' bytecodes should be included in the `factory_deps` as well.
 
-All the deployment process is handled inside our [hardhat](../../../api/hardhat) plugin.
+The [hardhat-zksync-deploy](../../../api/hardhat) plugin takes care of the deployment process. Here's a [guide on how to use it](../../../api/hardhat/getting-started.md).
 
 ## Solidity/Vyper support
 
-Compiling Solidity to zkEVM bytecode requires a special compiler. For the time being Solidity `>=0.4.0` versions are supported, though it is still recommended to use `^0.8.0` as the most stable ones. Vyper `^0.3.3` is also supported.
+Compiling Solidity to zkEVM bytecode requires a special compiler. For the time being Solidity `>=0.4.10` versions are supported, though we strongly recommended using `^0.8.0` as the most stable one. Vyper `^0.3.3` is also supported.
+
+Although, older versions of Solidity are supported, here are some of their limitations in zkSync:
+- Contract-local recursion is not supported.
+- Internal function pointers are not supported. 
+
+For smart contract compilation using Solidity or Vyper, [check here for their plugin](https://github.com/matter-labs/hardhat-zksync).
```

Comment:
I think the call to action should be "check the correspondent Hardhat plugins here", and the link should point to "api/hardhat/plugins.md"